### ISSUE TITLE
Refrain Nemo-Guardrails to Send the Actual User Input to LLM

### ISSUE THREAD

**Issue by minghongg** [NVIDIA/NeMo-Guardrails#706]
Hi team,

Is it possible to configure Nemo-Guardrails to avoid sending the actual user input to the LLM? I understand that the actual user input won't be sent if the input rails are triggered. However, is it also possible to prevent the user input from being sent, regardless of whether the input rails are triggered or not? Thanks!

**Comment by Drewwb** [comment]
Yes, it is possible to configure NeMo Guardrails to avoid sending the actual user input to the LLM, regardless of whether input rails are triggered. One effective approach to achieve this is:

- Use a secondary LLM as a pre-processing step. I would call this an Observer.
- This secondary LLM evaluates the user input against the defined guardrails.
- It outputs a simple "pass" or "fail" result.
- If the input passes, a sanitized or rephrased version of the input (not the original) is sent to the primary LLM. Or even "pass" is sent to LLM.
- If it fails, the input is blocked entirely.

In essence, you would prompt your Observer LLM in a way that makes sure it doesn't output the user's input.

**Comment by Pouyanpi** [comment]
@minghongg, do you mean that you want to use predefined flows only?

What do you want to do with the user input? It'd be great if you can explain your use case more

**Comment by drazvan** [comment]
Addi this for reference as well: 
https://docs.nvidia.com/nemo/guardrails/user_guides/input_output_rails_only/README.html#using-only-input-and-output-rails .

**Comment by Pouyanpi** [comment]
@minghongg did any of the answers resolved your issue? If yes, please close it.
